utils/env.go

```diff
@@ -0,0 +1,55 @@
+package oyster_utils
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	raven "github.com/getsentry/raven-go"
+)
+
+type Env struct {
+	ETH_NODE_URL         string
+	HOST_IP              string
+	MAIN_WALLET_ADDRESS  string
+	MAIN_WALLET_KEY      string
+	MODE                 string
+	SEGMENT_WRITE_KEY    string
+	SENTRY_DSN           string
+	TEST_MODE_WALLET_KEY string
+}
+
+var ENV = Env{
+	ETH_NODE_URL:         os.Getenv("ETH_NODE_URL"),
+	HOST_IP:              resolveHostIp(),
+	MAIN_WALLET_ADDRESS:  os.Getenv("MAIN_WALLET_ADDRESS"),
+	MAIN_WALLET_KEY:      os.Getenv("MAIN_WALLET_KEY"),
+	MODE:                 os.Getenv("MODE"),
+	SEGMENT_WRITE_KEY:    os.Getenv("SEGMENT_WRITE_KEY"),
+	SENTRY_DSN:           os.Getenv("SENTRY_DSN"),
+	TEST_MODE_WALLET_KEY: os.Getenv("TEST_MODE_WALLET_KEY"),
+}
+
+func resolveHostIp() string {
+	envHost := os.Getenv("HOST_IP")
+	if envHost != "" {
+		return envHost
+	}
+
+	netInterfaceAddresses, err := net.InterfaceAddrs()
+	if err == nil {
+		for _, netInterfaceAddress := range netInterfaceAddresses {
+			networkIp, ok := netInterfaceAddress.(*net.IPNet)
+			if ok && !networkIp.IP.IsLoopback() && networkIp.IP.To4() != nil {
+				ip := networkIp.IP.String()
+				fmt.Println("Resolved Host IP: " + ip)
+				return ip
+			}
+		}
+	}
```

Comment:
My only concern here would be, is this logic going to get the ip address of the actual host machine or is it going to be the ip address of the docker container?

Reply:
Yes, I forgot about that....I'm going to put a pause on this PR until we can think of a better solution. Otherwise we can just stick with a manual solution for now.